instagram-app/src/components/PostContainer/PostPage.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+//import withAuthenticate from './components/Authentication/withAuthenticate';
+import '../../App.css';
+import SearchBar from '../SearchBar/SearchBar';
+import dummyData from '../../dummy-data';
+import PostContainer from './PostContainer';
+
+class PostPage extends Component {
+  constructor() {
+    super();
+    this.state = {
+      data: [],
+      comments: {
+        text: '',
+        username: ''
+      }
+     
+    }
+  }
+  
+
+  addComment = (e) => {
+    e.preventDefault();
+    const thisIndex = e.target.getAttribute('index')
+    //console.log(this.state.data[thisIndex].comments)
+    this.state.data[thisIndex].comments.push({ text: this.state.comments.text, username: localStorage.getItem('user')})
+    this.setState({
+      data: [...this.state.data ]
```

Comment:
State values (like `this.state.data`) shouldn't be modified directly - it is best practice to use `.setState` instead for changes.

To avoid modifying the `data` state value directly, you can copy it to a new variable first and then use that new variable's value in `this.setState`, like so:
```
const newData = this.state.data.slice(); // The .slice() method (without arguments) also creates a deep copy of an array
newData[thisIndex].comments.push({ text: this.state.comments.text, username: this.state.comments.username});
this.setState({
  data: [...newData ]
});
```